lms/services/course.py

```diff
@@ -72,16 +72,16 @@ def get_from_launch(self, product, lti_params):
             copied_from=historical_course,
         )
 
-    def search(  # noqa: PLR0913, PLR0917
+    def _search_query(  # noqa: PLR0913, PLR0917
         self,
         id_: int | None = None,
         context_id: str | None = None,
         h_id: str | None = None,
         name: str | None = None,
-        limit: int = 100,
+        limit: int | None = 100,
```

Comment:
We were already using limit=None, fix the annotation.

LIMIT NULL means no limit.